objectPerformance/src/plotter.py

```diff
@@ -247,29 +246,126 @@ def _compute_value_of_tanh_at_threshold(self,
                                             efficiency,
                                             bins,
                                             threshold):
-        xvals = [x - (bins[1] - bins[0]) / 2 for x in bins[1:]]
-        popt, pcov = curve_fit(utils.tanh, xvals, efficiency, p0=[1,0])
+        xvals = bins
+        xvals = np.array(xvals); efficiency = np.array(efficiency)
+
+        xvals = xvals[~np.isnan(efficiency)]
+        efficiency = efficiency[~np.isnan(efficiency)]
+
+        popt, pcov = curve_fit(utils.tanh, xvals, efficiency)
         if np.inf in pcov:
             return None
 
         s_val = utils.arctanh(threshold, *list(popt))
 
         return s_val
 
+    def tridiagonal(self, H, K1, K2):
+
+        A = []
+        B = []
+        C = []
+        D = []
+
+        for i in range(len(K1)):
+            A.append(-K2)
+            B.append(K1[i]+2*K2)
+            C.append(-K2)
+
+        A[0] = 0
+        C[-1] = 0
+
+        for i in range(len(K1)):
```

Comment:
Why is this done in a separate loop? could just be done above, right?

Reply:
However, the way to do this nicely would be entirely without loops, but with numpy array slicing.